From b71ec21bb88885ec34a44daec9c6b9213a3f3e44 Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Fri, 25 Jul 2025 16:02:20 +0800
Subject: [PATCH 2/2] configurations: harma: add GPIO bridge for battery
 reading
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Change battery reading path from ADC sensor to PSU sensors
due to hardware design in 2nd source.

Tested:
xyz.openbmc_project.Sensor.Value interface - -
.MaxValue                        property  d 255
.MinValue                        property  d 0
.Unit                            property  s "xyz.openbmc_project.
                                              Sensor.Value.Unit.…
.Value                           property  d 3.036

Change-Id: I1c844da959f37849fc56d26b567e89b619fc1460
Signed-off-by: Peter Yin <peter.yin@quantatw.com>
---
 configurations/meta/harma_mb_vr_infineon.json | 38 +++++++++----------
 1 file changed, 18 insertions(+), 20 deletions(-)

diff --git a/configurations/meta/harma_mb_vr_infineon.json b/configurations/meta/harma_mb_vr_infineon.json
index 79cb339a2a..b447957bbc 100644
--- a/configurations/meta/harma_mb_vr_infineon.json
+++ b/configurations/meta/harma_mb_vr_infineon.json
@@ -2,20 +2,30 @@
     {
         "Exposes": [
             {
+                "Address": "0x1d",
                 "BridgeGpio": [
                     {
+                        "Label": "in0",
                         "Name": "rtc-battery-voltage-read-enable",
                         "Polarity": "High",
-                        "SetupTime": 1
+                        "SetupTime": 2
                     }
                 ],
-                "Index": 7,
-                "Name": "MB_P3V_BAT_VOLT_V",
-                "ScaleFactor": 0.333333333,
+                "Bus": 29,
+                "Labels": [
+                    "in0",
+                    "in1",
+                    "in2",
+                    "in3",
+                    "in4",
+                    "in5"
+                ],
+                "Name": "MB_ADC0",
                 "Thresholds": [
                     {
                         "Direction": "less than",
                         "Hysteresis": 0.052,
+                        "Label": "in0",
                         "Name": "lower critical",
                         "Severity": 1,
                         "Value": 2.6
@@ -23,25 +33,11 @@
                     {
                         "Direction": "greater than",
                         "Hysteresis": 0.068,
+                        "Label": "in0",
                         "Name": "upper critical",
                         "Severity": 1,
                         "Value": 3.4
-                    }
-                ],
-                "Type": "ADC"
-            },
-            {
-                "Address": "0x1d",
-                "Bus": 29,
-                "Labels": [
-                    "in1",
-                    "in2",
-                    "in3",
-                    "in4",
-                    "in5"
-                ],
-                "Name": "MB_ADC0",
-                "Thresholds": [
+                    },
                     {
                         "Direction": "less than",
                         "Hysteresis": 0.0209,
@@ -124,6 +120,8 @@
                     }
                 ],
                 "Type": "ADC128D818",
+                "in0_Name": "MB_P3V_BAT_VOLT_V",
+                "in0_Scale": 333.333333,
                 "in1_Name": "MB_PVDD11_S3_VOLT_V",
                 "in1_PowerState": "On",
                 "in2_Name": "MB_P3V3_STBY_VOLT_V",
-- 
2.25.1

